routers/repo/branch.go

```diff
@@ -230,6 +295,7 @@ func loadBranches(ctx *context.Context) []*Branch {
 			CommitsAhead:      divergence.Ahead,
 			CommitsBehind:     divergence.Behind,
 			LatestPullRequest: pr,
+			MergeMovedOn:      mergeMovedOn,
```

Comment:
Maybe this is an edge case and is covered elsewhere but, perhaps if we've got `ErrReferenceNotFound` we should hide the `new PR` button as well?

Reply:
Is this referring to when I thought it was possible to not get the head branch commit id? We actually always know that's there because otherwise there would have been a ServerError at line 200. So that can't happen and I've optimised that check away.

OR

is this referring to if we can't find the PR pulls head in the base repo?

I'm not sure what to do in that case. If I try to enforce that the pulls head is actually there I suspect that will break a lot of integration tests - as our fixtures do not necessarily match the ground truth. I also expect that there is a race between the db and the presence of the pull head. (We should check what happens here - the index is needed to create the pull head - so that means that the db has to have a pull request committed before we can push to the pull head.)

Reply:
I meant that `MergeMovedOn`, which has a restrictive meaning, is only set to true if both heads are found, but the restriction (which is to hide the `New PR` button) still should apply if any of the heads is missing (`ErrReferenceNotFound`). Maybe when any of  the heads is missing it should also be set to true, but then its name should be changed because it will no longer reflect the exact scenario.

Reply:
I don't understand.

Perhaps some of what you're discussing is already fixed within the logic of the template?

If you don't have a PR already then merge moved on is of no consequence.

The head branch we already know exists because it's the branch we've provided.

If the base branch does not exist - that's a server error - although this won't show that. You shouldn't be able to delete a branch with PRs against it - (although I haven't checked that.) If the git pull head isn't there well... that's technically a server error but I think there could be a race between between a pull request being made and the ref being updated so I think we should allow it.